packages/roosterjs-editor-plugins/lib/plugins/TableSelection/TableCellSelection.ts

```diff
@@ -0,0 +1,850 @@
+        const { which, shiftKey } = event.rawEvent;
+
+        if (which == RIGHT_CLICK && this.tableSelection) {
+            //If the user is right clicking To open context menu
+            const td = this.editor.getElementAtCursor(TABLE_CELL_SELECTOR);
+            if (td?.classList.contains(TABLE_CELL_SELECTED)) {
+                this.firstTarget = null;
+                this.lastTarget = null;
+
+                this.editor.queryElements('td.' + TABLE_CELL_SELECTED, node => {
+                    this.firstTarget = this.firstTarget || node;
+                    this.lastTarget = node;
+                });
+                const selection = this.editor.getDocument().defaultView.getSelection();
+                selection.setBaseAndExtent(this.firstTarget, 0, this.lastTarget, 0);
+                this.vTable.highlight();
+                return;
+            }
+        }
+        this.editor.getDocument().addEventListener('mouseup', this.onMouseUp, true /*setCapture*/);
+        if (which == LEFT_CLICK && !shiftKey) {
+            this.clearState();
+            this.editor
+                .getDocument()
+                .addEventListener('mousemove', this.onMouseMove, true /*setCapture*/);
+            this.startedSelection = true;
+        }
+    }
+
+    private handleMouseUpEvent(event: PluginMouseUpEvent) {
+        this.range = this.editor.getSelectionRange();
+        if (
+            event.isClicking &&
+            this.range.collapsed &&
+            !this.tableSelection &&
+            event.rawEvent.which != RIGHT_CLICK
+        ) {
+            this.clearTableCellSelection();
+        }
+    }
+
+    private onMouseMove = (event: MouseEvent) => {
+        if (event.currentTarget == this.contentDiv) {
+            return;
+        }
+
+        this.setData(event.target as Node);
+        //Ignore if
+        // Is a DIV that only contains a Table
+        // If the event target is not contained in the editor.
+        if (
+            (this.lastTarget.lastChild == this.lastTarget.firstChild &&
+                getTagOfNode(this.lastTarget.lastChild) == 'TABLE' &&
+                getTagOfNode(this.lastTarget) == 'DIV') ||
+            !this.contentDiv.contains(this.lastTarget)
+        ) {
+            event.preventDefault();
+            return;
+        }
+
+        this.prepareSelection();
+
+        const targetIsValid = this.targetTable == this.firstTable;
+        const isNewTDContainingFirstTable = safeInstanceOf(this.lastTarget, 'HTMLTableCellElement')
+            ? contains(this.lastTarget, this.firstTable)
+            : false;
+
+        if (this.firstTable && (targetIsValid || isNewTDContainingFirstTable)) {
+            //When starting selection inside of a table and ends inside of the same table.
+            this.selectionInsideTableMouseMove(event);
+        } else if (this.tableSelection) {
+            clearSelectedTableCells(this.contentDiv);
+            this.tableSelection = false;
```

Comment:
sees some of the code can be replaced with editor.getSelectedRegions()

Reply:
Added getSelectedRegions to the code as recommended.